src/core/trulens/core/instruments.py

```diff
@@ -133,7 +137,7 @@ def on_new_record(self, func: Callable):
     # Called during invocation.
     def on_add_record(
         self,
-        ctx: RecordingContext,
+        ctx: "RecordingContext",
```

Comment:
When do we use a string vs Type.

Reply:
In this case, importing `RecordingContext` would cause a circular import. Python has an option to get around this by importing under an `if typing.TYPE_CHECKING` condition (see line 63). code and imports here are only run for type checking and are ignored at runtime. Types referencing imports under `typing.TYPE_CHECKING` are stringified. Most common IDEs and type checkers are able to match it to the appropriate type.